### PR TITLE
[CUREPIC-86] User facing curriculum downloads 

### PR DESCRIPTION
**Pull request by orangemug**
This is the feature for user facing downloads of curriculum documents in the docx format.

Features

 - Generates documents for all supported curriculum in the docx format
 - Adds XML formatting via prettier inside XML strings
 - Adds cached API route for downloading documents (cached via netlify)
 - New "downloads" tab in the curriculum visualiser

https://github.com/user-attachments/assets/f223211a-ad68-4212-b81f-9b4f4b2b5994
  

#### Testing
To test

1. Go to https://deploy-preview-2552--oak-web-application.netlify.thenational.academy/teachers/curriculum/docx-poc/select
2. Type a subject name in the search field
3. Choose exam board where necessary, and select 'published' version for cycle 1 lots or 'new' for cycle 2
4. Click 'Generate Document' button

OR

1. Load up the [Curriculum Landing Page](https://deploy-preview-2552--oak-web-application.netlify.thenational.academy/teachers/curriculum) for this branch
2. Open up DevTools (In Mac Chrome it's `Cmd + Option + I`)
3. In the Application tab, select `Cookies` from the sidebar then click on the `https://deploy-preview-2552[...]` entry in the sub-menu.
4. Create a new cookie with a name of `prerelease.curriculum.downloads.enabled` and a value of `true`
5. Navigate to any curriculum lot e.g. [Secondary Science AQA](https://deploy-preview-2552--oak-web-application.netlify.thenational.academy/teachers/curriculum/science-secondary-aqa/downloads)
6. Select the 'Download' tab
7. Fill in your details then click the Download button

### CODE REVIEW

**Review by sonalisin — APPROVED** [review]
LGTM 👍

File: src/pages-helpers/curriculum/docx/builder/helper.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  return value !== undefined;
+}
+
+export function createCurriculumSlug(slugs: Slugs) {
+  return `${slugs.subjectSlug}-${slugs.phaseSlug}${
+    slugs.examboardSlug ? `-${slugs.examboardSlug}` : ""
+  }`;
+}
+
+// TODO: This is from from Sonali's work that's not yet merged
```
Comment (orangemug) [inline src/pages-helpers/curriculum/docx/builder/helper.ts:71]: This is dup code and should be removed and use the in existing version instead

File: src/components/CurriculumComponents/CurriculumDownloadTab/index.tsx [hunk 2]
```diff
@@ -65,22 +145,75 @@ const CurriculumDownloadTab: FC = () => {
...
+    };
+    saveDownloadsDataToLocalStorage(schoolData);
 
-    // TODO: Actually generate file here (async)
-
-    setIsSubmitting(false);
-    setIsDone(true);
+    try {
+      await downloadFileFromUrl(downloadPath);
+      // TODO: Log to hubspot here...
```
Comment (orangemug) [inline src/components/CurriculumComponents/CurriculumDownloadTab/index.tsx:212]: This is done elsewhere now remove comment

File: src/pages-helpers/curriculum/docx/builder/helper.ts (hunk 1 above)
Comment (orangemug) [reply src/pages-helpers/curriculum/docx/builder/helper.ts]: Fixed in https://github.com/oaknational/Oak-Web-Application/pull/2552/commits/b054774714824e83b2c202da309afc89027745dc

**Review by sonalisin — APPROVED** [review]

**Review by rightsaidjames — APPROVED** [review]
👩‍🍳 💋 

LGTM 🎉

**Pull request merged**